### ISSUE TITLE
potential one-hot implementation

### ISSUE THREAD

**Issue by kaczmarj** [p-koo/cipher#44]
jax has an example of a one-hot implementation: https://jax.readthedocs.io/en/latest/notebooks/neural_network_with_tfds_data.html#utility-and-loss-functions

```python
def one_hot(x, k, dtype=jnp.float32):
  """Create a one-hot encoding of x of size k."""
  return jnp.array(x[:, None] == jnp.arange(k), dtype)
```

we could try a similar implementation. might be faster than the current version
